woocommerce/woocommerce
# Update refresh button to use API

### Changes proposed in this Pull Request:

The refresh button works by redirecting to the old subscriptions page and back. We need to replace this with API calls before we can remove the old subscriptions page.

This PR implements a new button that uses the `/refresh` API endpoint to upload the subscriptions context with latest data from Woo.com.

The new button also has a busy state to indicate that it's working.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open the new subscriptions page and connect to Woo.com if you are not connected
2. Click the refresh button
3. The button should be set to busy
4. Confirm that it sent an API request to `/wc/v3/marketplace/refresh`
5. When the request finishes, the button should stop spinning and a notice will appear



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

Subscriptions refresh button uses the API instead of redirect.

</details>
